### PR TITLE
Document the learning from controller-runtime bump

### PR DESCRIPTION
**Pull request by gibizer**

### CODE REVIEW

File: bumping_dependencies.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+1.20.
+
+1. Before moving to a new golang version you need to ensure that
+`ubi9/go-toolset` container available with the given golang version.
+
+2. Due to our upstream build logic you need to land all the version bumps in
+lib-common and the service operators first. Then you need to bump the the
+dependencies in openstack-operator as a last step. The CI will be red until to
+land the openstack-operator change. So agree on a day (or two) with the
+community when such CI outage is acceptable.
```
[the remark this answers is not in the document]
Comment (bshephar) [reply bumping_dependencies.md:34]: Or just not having to actually merge things until they are green. Whatever that might look like, I guess we would need a more complex system of Depends-On to accomplish that.

File: bumping_dependencies.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+[ci-framework](https://github.com/openstack-k8s-operators/ci-framework/commit/5c138d4f734e600cde2e78859da21ab9fa835d33)
+is using the newer golang version as well.
+
+After the last openstack-operator change is landed and CI is back to green
+there are some follow ups you can do. (Some can be done during the original
+bump if you are feeling lucky):
+
+1. Move the version pins in the
+[global renovate config](https://github.com/openstack-k8s-operators/renovate-config/commit/f25b1bb774777111a4ecf4829532e5840176971a)
+to the new maximums.
```
Comment (gibizer) [inline bumping_dependencies.md:62]: also if there was a golang version bump then change the go constraint in the global config.

File: bumping_dependencies.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+2. Due to our upstream build logic you need to land all the version bumps in
+lib-common and the service operators first. Then you need to bump the the
+dependencies in openstack-operator as a last step. The CI will be red until to
+land the openstack-operator change. So agree on a day (or two) with the
+community when such CI outage is acceptable.
+
+3. Even if the kuttl and zuul based CI will be red on the service operator in
+the meantime you can still run the envtest based functional tests to have some
+level of confidence about the changes. If `/api` changes are involved you can
+user `go.mod` replace lines pointing to unmerged dependencies to run functional
```
Comment (bogdando) [inline bumping_dependencies.md:39]: s/user/edit/ ?

File: bumping_dependencies.md (hunk 2 above)
Comment (bogdando) [reply bumping_dependencies.md:62]: could you post an example snippet as well?

**Comment by gibizer** [comment]
I almost forgot about this, but now I fixed the comments and I think it make sense to land this.

**Pull request merged**